### ISSUE TITLE
`Locale.localize()` now optional, is that correct?

### ISSUE THREAD

**Issue by sbusch** [date-fns/date-fns#1589]
Due to #1542, I cannot rely on `localize()` function being present on the Locale object, which forced me to change my implementation:

```diff
-export const weekdayNumberToGermanText = (weekdayNumber: number) =>
-  de.localize.day(weekdayNumber);
+export const weekdayNumberToGermanText = (weekdayNumber: number) => {
+  if (!de.localize) {
+    throw new Error('de.localize() missing');
+  }
+  return de.localize.day(weekdayNumber);
+};
```

Is it correct that `localize()` is not present on all Locale implementations? The change needs to be reverted for all mandatory Locale properties (if any).

For functions like `formatDistance` accepting partial Locale objects, IMO *their* interface should express that (e.g. like `Pick<Locale, 'formatDistance'>`).

**Comment by kossnocorp** [comment]
Sorry, that was an attempt to fix [#1503](https://github.com/date-fns/date-fns/issues/1503). We actually require all locales to have every field unless you use a custom one. Sadly I didn't anticipate this issue otherwise I would not merge [the PR](https://github.com/date-fns/date-fns/pull/1542). In the next version, we plan to migrate to TypeScript so all TS-related problems will go away.

**Comment by davidspiess** [comment]
Same problem. No chance to get a bugfix release until the TS release is ready?
